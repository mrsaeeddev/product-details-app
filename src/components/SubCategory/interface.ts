export interface subcategory {
    subcategory: string,
    handlePress(v: string): void,
    handleAllProducts(): void,
}
