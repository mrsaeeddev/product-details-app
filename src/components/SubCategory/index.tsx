import React from "react";
import { View, Text, StyleSheet } from "react-native";
import { widthPercentageToDP as wp } from 'react-native-responsive-screen';
import { globalStyles } from "../../globalStyles";

const styles = StyleSheet.create({
    inactive: {
        width: wp('10%'),
        height: 50,
        backgroundColor: 'lightgray',
        resizeMode: 'stretch',
        borderRadius: 10,
    },
});

function SubCategory() {
    return (<View style={[styles.inactive, globalStyles.alignItemsCenter]}><Text style={globalStyles.textCenter}>Saeed</Text></View>)
}

export default SubCategory;
