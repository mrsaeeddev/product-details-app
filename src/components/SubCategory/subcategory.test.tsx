import React from 'react';
import renderer from 'react-test-renderer';

import SubCategory from './';

describe('SubCategory', () => {
    it('should render correctly using Snapshot', () => {
        const subcategory = 'dummy1'
        const handlePress = jest.fn()
        const handleAllProducts = jest.fn()
        const tree = renderer.create(<SubCategory subcategory={subcategory} handlePress={handlePress} handleAllProducts={handleAllProducts} />).toJSON();
        expect(tree).toMatchSnapshot();
    });
});