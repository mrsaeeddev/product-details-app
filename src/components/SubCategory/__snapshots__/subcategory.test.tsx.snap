// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubCategory should render correctly using Snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "flex-start",
          "backgroundColor": "#F5F5F5",
          "borderRadius": 10,
          "height": 50,
          "marginRight": 15,
          "resizeMode": "stretch",
          "width": 225,
        },
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "textAlign": "center",
          },
          Object {
            "fontWeight": "bold",
          },
        ]
      }
    >
      dummy1
    </Text>
  </View>
</View>
`;
